File: src/ai/fallback-parser.ts
import { ParsedIntent, IntentStatus } from './types.js';

export class FallbackParser {
  static parse(prompt: string): ParsedIntent | null {
    const normalized = prompt.trim().toLowerCase();

    const createChannelMatch = normalized.match(/create (?:a )?(?:text )?channel (?:called |named )?([a-z0-9-_]+)/i);
    if (createChannelMatch) {
      return {
        status: IntentStatus.DIRECT_ACTION,
        action: 'create_channel',
        parameters: {
          name: createChannelMatch[1],
          type: 'text',
        },
      };
    }

    const sendMessageMatch = normalized.match(/send (?:message )?"([^"]+)" to ([a-z0-9-_]+)/i);
    if (sendMessageMatch) {
      return {
        status: IntentStatus.DIRECT_ACTION,
        action: 'send_message',
        parameters: {
          content: sendMessageMatch[1],
          channelName: sendMessageMatch[2],
        },
      };
    }

    return null;
  }
}

File: bot-py/src/ai/fallback_parser.py
import re
from typing import Any


class FallbackParser:
    @staticmethod
    def parse(prompt: str) -> dict[str, Any] | None:
        normalized = prompt.strip().lower()

        # Match "create channel <name>"
        match = re.search(r"create (?:a )?(?:text )?channel (?:called |named )?([a-z0-9-_]+)", normalized)
        if match:
            return {
                "action": "create_channel",
                "parameters": {"name": match.group(1), "type": "text"},
            }

        # Match "delete channel <id>"
        match = re.search(r"delete channel ([0-9]+)", normalized)
        if match:
            return {
                "action": "delete_channel",
                "parameters": {"channel_id": match.group(1)},
            }

        return None
